types/zookeeper/index.d.ts

```diff
@@ -0,0 +1,108 @@
+// Type definitions for zookeeper 3.4
+// Project: https://github.com/yfinkelstein/node-zookeeper#readme
+// Definitions by: xialeistudio <https://github.com/xialeistudio>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+/// <reference types="node" />
+
+type ACL = number | { perms: number, scheme: string, auth: string };
+
+interface Stat {
+    czxid: number;
+    mzxid: number;
+    ctime: number;
+    mtime: number;
+    version: number;
+    cversion: number;
+    aversion: number;
+    ephemeralOwner: string;
+    dataLength: number;
+    numChildren: number;
+    pzxid: number;
+}
+
+interface ConnectionOptions {
+    connect?: string;
+    timeout?: number;
+    host_order_deterministic?: boolean;
+    data_as_buffer?: boolean;
+    debug_level?: number;
+}
+
+type path_callback = (rc?: number, error?: string, path?: string) => void;
```

Comment:
This is only used once. It would be better to just write it inline.
Of course, you might also make use of:
```ts
type Callback<T> = (rc?: number, error?: string, value?: T) => void;
```
then
```ts
a_create(path: string, data: string | Buffer, flags: number, callback: Callback<string>): void;
```

Another thing to note: Parameters in callbacks should only be marked optional if they are not always *provided*. It's always free to not *use* a parameter and it doesn't have to be marked optional for that. See [Optional Parameters in Callbacks](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html).

Reply:
Thank you for your patience.It's very helpful to me.